tensorboard/plugins/histogram/histograms_plugin.py

```diff
@@ -51,16 +68,62 @@ def is_active(self):
     return bool(self._multiplexer) and any(self.index_impl().values())
 
   def index_impl(self):
-    return {
-        run_name: run_data[event_accumulator.HISTOGRAMS]
-        for (run_name, run_data) in self._multiplexer.Runs().items()
-        if event_accumulator.HISTOGRAMS in run_data
-    }
+    """Return {runName: {tagName: {displayName: ..., description: ...}}}."""
+    runs = self._multiplexer.Runs()
+    result = {run: {} for run in runs}
+
+    # Old-style runs
+    for (run, run_data) in six.iteritems(runs):
+      for tag in run_data.get(event_accumulator.HISTOGRAMS, []):
```

Comment:
A single run could indeed have both old-style and new-style tags. I imagine that it will be common for users to have old code (copied from an old tutorial, written by the user or a friend a while ago, etc.) with old-style summaries, and then add some new-style summaries to the same code. Such a run would have both kinds of tags and should work just fine. (A single _tag_ should never occur twice.)